src/jit/optimizer.cpp

```diff
@@ -1148,20 +1141,29 @@ bool Compiler::optRecordLoop(BasicBlock*   head,
     assert(entry->bbNum <= bottom->bbNum);
     assert(head->bbNum < top->bbNum || head->bbNum > bottom->bbNum);
 
-    // If the new loop contains any existing ones, add it in the right place.
     unsigned char loopInd = optLoopCount;
-    for (unsigned char prevPlus1 = optLoopCount; prevPlus1 > 0; prevPlus1--)
+
+    if (optLoopTable == nullptr)
     {
-        unsigned char prev = prevPlus1 - 1;
-        if (optLoopTable[prev].lpContainedBy(first, bottom))
-        {
-            loopInd = prev;
-        }
+        assert(loopInd == 0);
+        optLoopTable = static_cast<LoopDsc*>(compGetMemArray(MAX_LOOP_NUM, sizeof(LoopDsc), CMK_LoopOpt));
     }
-    // Move up any loops if necessary.
-    for (unsigned j = optLoopCount; j > loopInd; j--)
+    else
     {
-        optLoopTable[j] = optLoopTable[j - 1];
+        // If the new loop contains any existing ones, add it in the right place.
+        for (unsigned char prevPlus1 = optLoopCount; prevPlus1 > 0; prevPlus1--)
+        {
+            unsigned char prev = prevPlus1 - 1;
+            if (optLoopTable[prev].lpContainedBy(first, bottom))
+            {
+                loopInd = prev;
+            }
+        }
+        // Move up any loops if necessary.
+        for (unsigned j = optLoopCount; j > loopInd; j--)
+        {
+            optLoopTable[j] = optLoopTable[j - 1];
+        }
```

Comment:
Is this an actual code behavior change?
It would be best to keep this checkin to things that don't cause any behavior changes.
And use a seperate change for anything that could cause some code diffs. (Even if it is a bug fix)

Reply:
Hmm, there should not be any behavior change. Are you referring to the fact that this loop no longer executes when the loop table is nullptr? That's because there's no loop in the table.

Reply:
It looked like your were adding new code here, but if you are just reordering stuff that is OK.